src/mydumper_common.c

```diff
@@ -548,7 +601,10 @@ void initialize_sql_statement(GString *statement){
   if (is_mysql_like())  {
     if (set_names_statement)
       g_string_printf(statement,"%s;\n",set_names_statement);
+    g_string_append(statement, "/*!40014 SET UNIQUE_CHECKS=0*/;\n");
```

Comment:
Why should this be set conditionally? What's the point in UNIQUE_CHECKS=1?

Reply:
How it can not fail with FOREIGN_KEY_CHECKS=1? Do you track correct table creation order?

Reply:
The question stays: why do you want them set conditionally if it does not make sense?

Reply:
No, I'm asking specifically about FOREIGN_KEY_CHECKS and UNIQUE_CHECKS. Are there some DB vendors that require other values for these variables? UNIQUE_CHECKS should be 1 for both MariaDB and MySQL. FOREIGN_KEY_CHECKS=1 doesn't work at all: what's the point having it in config file? Will it fail without config file? I compile code from scratch and start it without installation, so I don't use any config file.

Reply:
What about SQL_NOTES and NO_AUTO_VALUE_ON_ZERO. I guess NO_AUTO_VALUE_ON_ZERO should be there unconditionally, otherwise it produces different result on restored database.

Reply:
SQL_NOTES can be added to the default configuration file. 
We are setting NO_AUTO_VALUE_ON_ZERO to sql_mode in the default configuration file. 

take into account that nowadays it is really important the configuration file, which is loaded by default from /etc/mydumper.cnf when you install it. However, when you compile MyDumper, it is not copy to the default location, that is why it is not loaded. That is why, I recommend you to compile and install if you are going to perform tests.

Reply:
The software should work correctly without any configuration files. Different result between the origin and the copy should be considered as bug. Default normally means something that doesn't require any explicit configuration settings. Of course, this patch produces double directives over configuration, so it is not correct too. I will look if it is possible to make default configuration layer in the code.

Reply:
I updated this change to commit 7e76589